src/classes/Block.ts

```diff
@@ -0,0 +1,179 @@
+  }
+
+  private _registerLifecycleEvents(eventBus: EventBus) {
+    eventBus.on(Block.EVENTS.INIT, this._init.bind(this));
+    eventBus.on(Block.EVENTS.FLOW_CDM, this._componentDidMount.bind(this));
+    eventBus.on(Block.EVENTS.FLOW_CDU, this._componentDidUpdate.bind(this));
+    eventBus.on(Block.EVENTS.FLOW_RENDER, this._render.bind(this));
+  }
+
+  private _createResources() {
+    const { tagName } = this._meta;
+    this._element = this._createDocumentElement(tagName);
+  }
+
+  private _init() {
+    this._createResources();
+    this.eventBus().emit(Block.EVENTS.FLOW_CDM);
+  }
+
+  private _componentDidMount() {
+    this.componentDidMount();
+    this.eventBus().emit(Block.EVENTS.FLOW_RENDER);
+  }
+
+  public componentDidMount() { }
+
+  private _componentDidUpdate(oldProps: TProps, newProps: TProps) {
+    const response = this.componentDidUpdate(oldProps, newProps);
+    if (response) {
+      this.eventBus().emit(Block.EVENTS.FLOW_RENDER);
+    }
+  }
+
+  public componentDidUpdate(oldProps?: TProps, newProps?: TProps) {
+    return oldProps !== newProps;
+  }
+
+  public setProps = (nextProps: TProps) => {
+    if (!nextProps) {
+      return;
+    }
+    Object.assign(this.props, nextProps);
+  };
+
+  get element() {
+    return this._element;
+  }
+
+  private _addEvents() {
+    const { events = {} } = this.props;
+
+    Object.entries(events).forEach(([eventName, event]: [string, (evt: Event) => {}]) => {
+      this._element.addEventListener(eventName, event);
+    });
+  }
+
+  private _removeEvents() {
+    const { events = {} } = this.props;
+
+    Object.keys(events).forEach((eventName) => {
+      this._element.removeEventListener(eventName, events[eventName]);
+    });
+  }
+
+  private _renderChildComponents(elements: NodeListOf<Element>) {
+    elements.forEach((markerElement) => {
+      if (markerElement instanceof HTMLElement) {
+        const parent = markerElement.parentNode;
+        if (parent && markerElement.dataset.uuid) {
+          const blockElement = window._componentStore[markerElement.dataset.uuid];
+          parent.replaceChild(blockElement, markerElement);
+        }
+      }
+    });
+  }
+
+  private _render() {
+    const blockHTML = this.render();
+    this._removeEvents();
+    if (blockHTML) {
+      const template = getElementsFromString(blockHTML);
+      // Перенос атрибутов с <template> на обёртку блока
+      if (template) {
+        template.getAttributeNames()
+          .forEach((name) => {
+            this._element.setAttribute(name, template.getAttribute(name) || '');
+          });
+        const blockElements = template.content.cloneNode(true);
+        this._element.innerHTML = '';
+        this._element.append(blockElements);
+        const markerElements = this._element.querySelectorAll('[data-uuid]');
+        this._renderChildComponents(markerElements);
+        this._element.removeAttribute('data-uuid');
+      }
+    }
+    this._addEvents();
```

Comment:
Высокая вложенность, что повышает цикломатическую сложность этого кода. Проще говоря, его сложнее понять, потому что приходится держать в голове условия if 
Я бы переписал вот так
```suggestion
    if (!blockHTML && !template) {
      return
    }

      const template = getElementsFromString(blockHTML);
      // Перенос атрибутов с <template> на обёртку блока
        template.getAttributeNames()
          .forEach((name) => {
            this._element.setAttribute(name, template.getAttribute(name) || '');
          });
        const blockElements = template.content.cloneNode(true);
        this._element.innerHTML = '';
        this._element.append(blockElements);
        const markerElements = this._element.querySelectorAll('[data-uuid]');
        this._renderChildComponents(markerElements);
        this._element.removeAttribute('data-uuid');
    this._addEvents();
```